### PR TITLE
fifa-champs integration with slack bot @edsonbastos

### PR DESCRIPTION
**Pull request by talesprates**
this pull request is going to add the slack bot user named @edsonbastos that will enable admin users to run commands through fiao (or any other slack) slack chat. The only commands implemented will be add match, add final, and help.

### CODE REVIEW

File: api/dist [hunk 1]
```diff
@@ -0,0 +1 @@
+../webapp/dist
```
Comment (cristiadu) [inline api/dist]: why is this here?

File: api/integrations/fifabot/.eslintrc.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
+module.exports = {
```
Comment (cristiadu) [inline api/integrations/fifabot/.eslintrc.js]: can't we use the same lint we have for the api project? Why do we have a different config file for lint in here?

File: api/integrations/fifabot/commands/addMatch.js [hunk 3]
```diff
@@ -0,0 +1,57 @@
+const request = require('request-promise');
+
+const pattern = /^add match (.*) (.*) (\d+)\s*x\s*(\d+) (.*) (.*)$/;
+const seasonOptions = {
+  url: 'http://localhost:5001/api/championships',
```
Comment (cristiadu) [inline api/integrations/fifabot/commands/addMatch.js]: This will be changed later on to use the environment variables, right? Localhost isn't going to be the correct one.

Also, can't you use the baseAPI url and just concatenate with the specific endpoint later on?

File: api/integrations/fifabot/.gitignore [hunk 4]
```diff
@@ -0,0 +1,138 @@
+
```
Comment (cristiadu) [inline api/integrations/fifabot/.gitignore]: I would rather not have this generated gitignore here. Just create one with the files you actually need to ignore, not an giant set of files we will never use inside the repo.

File: api/integrations/fifabot/commands/addMatch.js [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+  url: 'http://localhost:5001/api/players',
+  method: 'GET',
+  json: true
+};
+const matchOptions = {
+  url: 'http://localhost:5001/api/matches',
+  method: 'POST',
+  json: true
+};
+const PLAYER1 = 1;
```
Comment (cristiadu) [inline api/integrations/fifabot/commands/addMatch.js]: we don't use the index 0?

File: api/integrations/fifabot/commands/addMatch.js [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+    Promise.all([seasonsPromise, playersPromise])
+      .then(([seasonsBody, playersBody]) => {
+        const championship = seasonsBody.find(season => season.isCurrent === true);
+        const player1 = playersBody.find(player => player.nickname === args[PLAYER1]);
+        const player2 = playersBody.find(player => player.nickname === args[PLAYER2]);
+        const team1score = args[TEAM1SCORE];
+        const team2score = args[TEAM2SCORE];
+        const player3 = playersBody.find(player => player.nickname === args[PLAYER3]);
+        const player4 = playersBody.find(player => player.nickname === args[PLAYER4]);
+        const date = new Date().toISOString();
```
Comment (cristiadu) [inline api/integrations/fifabot/commands/addMatch.js]: Are we going to allow inputting the date as well? Or only games on the same day as the message?

File: api/integrations/fifabot/commands/addMatch.js [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+        return {
+          championship, player1, player2, player3, player4, team1score, team2score, isFinal, date
+        };
+      })
+      .then(match => request(Object.assign(matchOptions, { form: match })))
+      .then(() => resolve('partida adicionada.'))
```
Comment (cristiadu) [inline api/integrations/fifabot/commands/addMatch.js]: Well, we have to decide. Or we have all in Portuguese, or all in English.

File: api/integrations/fifabot/commands/addMatch.js [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+      .catch(error => reject(`partida não adicionada (${error})`));
```
Comment (cristiadu) [inline api/integrations/fifabot/commands/addMatch.js]: Well, we have to decide. Or we have all messages in Portuguese, or all in English.

File: api/integrations/fifabot/integrations/slack.js [hunk 9]
```diff
@@ -0,0 +1,15 @@
+const Botkit = require('botkit');
```
Comment (cristiadu) [inline api/integrations/fifabot/integrations/slack.js:1]: I think we need a better name for this file. This is the setup file needed for all slack integrations configured, so we could rename as such.

File: api/integrations/fifabot/package.json [hunk 10]
```diff
@@ -0,0 +1,40 @@
+{
+  "name": "fifabot",
+  "version": "1.0.0",
+  "main": "fifabot.js",
+  "dependencies": {
+    "body-parser": "^1.17.1",
+    "botkit": "^0.4.10",
```
Comment (cristiadu) [inline api/integrations/fifabot/package.json]: Do we really need all those dependencies?

**Review by cristiadu — CHANGES_REQUESTED** [review]

File: api/integrations/fifabot/.eslintrc.js (hunk 2 above)
Comment (talesprates) [reply api/integrations/fifabot/.eslintrc.js]: yes we can, i'll see how i can configure that.

File: api/integrations/fifabot/.gitignore (hunk 4 above)
Comment (talesprates) [reply api/integrations/fifabot/.gitignore]: ok. i'll reset the gitignore.

File: api/integrations/fifabot/commands/addMatch.js (hunk 5 above)
Comment (talesprates) [reply api/integrations/fifabot/commands/addMatch.js]: index 0 is the full match from regex.

File: api/integrations/fifabot/commands/addMatch.js (hunk 6 above)
Comment (talesprates) [reply api/integrations/fifabot/commands/addMatch.js]: it is a good idea. i have to check with @alandoni and @kiosia how the date input works for the matches.

File: api/integrations/fifabot/commands/addMatch.js (hunk 7 above)
Comment (talesprates) [reply api/integrations/fifabot/commands/addMatch.js]: hum.. i was thinking about that all messages that goes to the users are portuguese and cli messages english.

File: api/integrations/fifabot/integrations/slack.js (hunk 9 above)
Comment (talesprates) [reply api/integrations/fifabot/integrations/slack.js:1]: botkit is a package that interfaces with slack API for us. what can be done is renamed just the variable, not the file

File: api/integrations/fifabot/package.json (hunk 10 above)
Comment (talesprates) [reply api/integrations/fifabot/package.json]: i have just copy pasted the dependencies from another project. i'll check again the list.

File: api/integrations/fifabot/integrations/slack.js (hunk 9 above)
Comment (cristiadu) [reply api/integrations/fifabot/integrations/slack.js:1]: I wasn't talking about the line itself, but the file name.

I meant that this is not an integration by itself, so it should be renamed. We can leave that as it is right now since I can't think of a better name ("slack_setup", "base_slack", or something like that, idk).
